src/main/java/com/redhat/jenkins/plugins/ci/threads/TriggerThreadTransientActionFactory.java

```diff
@@ -0,0 +1,32 @@
+package com.redhat.jenkins.plugins.ci.threads;
+
+import hudson.Extension;
+import hudson.model.AbstractProject;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import jenkins.model.TransientActionFactory;
+
+import com.redhat.jenkins.plugins.ci.CIBuildTrigger;
+
+@SuppressWarnings("rawtypes")
+@Extension
+public class TriggerThreadTransientActionFactory extends TransientActionFactory {
+
+    @Override
+    public Class type() {
+        return AbstractProject.class;
+    }
+
+    @Override
+    public Collection createFor(Object target) {
+        if (target instanceof AbstractProject) {
+            AbstractProject project = (AbstractProject)target;
+            @SuppressWarnings("unchecked")
+            CIBuildTrigger cibt = (CIBuildTrigger)project.getTrigger(CIBuildTrigger.class);
```

Comment:
Is there no way to implement this class in type-safe way? `ParameterizedJobMixIn.ParameterizedJob#getTriggers()` should help here with both type-safety and adding support for pipeline jobs as they are not `AbstractProject`s.

Reply:
@olivergondza , that's exactly why I want you reviewing changes like these....

Reply:
Done.